sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -882,19 +883,34 @@ struct _pi_queue : _pi_object {
     uint32_t NextIndex{0};
   };
 
+  // When using standard commandlists, a group containing compute queue handles.
+  // Also used for immediate commandlists when they are not thread-specific.
   pi_queue_group_t ComputeQueueGroup{this, queue_type::Compute};
 
-  // Vector of Level Zero copy command command queue handles.
-  // In this vector, main copy engine, if available, come first followed by
-  // link copy engines, if available.
+  // When using immediate command lists in per-thread mode, the QueueGroups are
+  // per thread, with the thread-id used as a key in this map. The single
+  // QueueGroup above, serves as an initial value to be used to initialize each
+  // new thread's QueueGroup.
+  std::unordered_map<std::thread::id, pi_queue_group_t> ComputeQueueGroupsByTID;
```

Comment:
I think we should just use the map for everything (and have a map of single element for non-per-thread case). Alternatively, if you are worried about map performance for some reason, we should not introduce this new map here, but have this into implementation detail of `pi_queue_group_t`